### PR TITLE
I6538 remove ab exp/logs from homepg

### LINKED ISSUE #6538: Remove withExperiment HOC from the HomeHeroBanner component + extra log calls
We should remove the `withExperiment` HOC from the `HomeHeroBanner` component as we do not run this experiment anymore. Extra log calls added in https://github.com/mozilla/addons-frontend/pull/6489 should be removed too, and we should also remove the config parameter.

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #6538

### CODE REVIEW

**Comment by willdurand** [comment]
Note: we are not ready to merge this PR since we did not find the root cause of the HomeHeroBanner issue.

**Comment by rebmullin** [comment]
hey @willdurand, what's the status on this one? Should I still keep this one open?

**Comment by willdurand** [comment]
I don't find anything in Kibana, so I guess the problem is somewhere else (React reconciliation?). I don't know what to do now since it was impossible to reproduce the bug in any env but -prod, and logs don't contain any weird behaviors...

@kumar303 what do you think? Do you want to take a closer look? Or should we remove the code (this PR) and see if the issue happens again during the next a/b experiment?

**Comment by willdurand** [comment]
@rebmullin we found the root cause of the issue. We need to fix it first and make sure it has been fixed entirely. So I think either you close this PR right now and re-open it later (after next week I believe), or you keep it as-is (and rebase regularly).

**Comment by willdurand** [comment]
> It would be hard to use our logs to determine that the `react-cookie` upgrade (#6803) fixes this bug.

Can't we make sure we won't see any logs with `cookie saved` twice?

**Comment by kumar303** [comment]
Yes, we could but I don't know how to search for that in kibana. I only found the dupes by scanning the first page of results with my eyes and I wouldn't want to do that across many pages.

**Comment by willdurand** [comment]
> Yes, we could but I don't know how to search for that in kibana. I only found the dupes by scanning the first page of results with my eyes and I wouldn't want to do that across many pages.

can we try that during the next tag?

**Comment by rebmullin** [comment]
hey @willdurand, this has been updated. Can you please see how it looks now?

File: src/amo/components/HomeHeroBanner/index.js [hunk 1]
```diff
@@ -314,27 +292,16 @@ export class HomeHeroBannerBase extends React.Component<InternalProps> {
...
 
-const HomeHeroBanner: React.ComponentType<InternalProps> = compose(
-  translate(),
-  withExperiment({
-    id: AB_HOME_HERO_EXPERIMENT,
-    variantA: AB_HOME_HERO_VARIANT_A,
-    variantB: AB_HOME_HERO_VARIANT_B,
-  }),
-)(HomeHeroBannerBase);
+const HomeHeroBanner: React.ComponentType<InternalProps> = compose(translate())(
```
Comment (willdurand) [inline src/amo/components/HomeHeroBanner/index.js]: You can get rid of `compose()` too and write: `... = translate()(HomeHeroBannerBase);`

File: src/core/withExperiment.js [hunk 2]
```diff
@@ -59,43 +59,18 @@ export const withExperiment = ({
...
-      });
-
-      log.info('[WithExperiment.constructor] this.props:', {
-        variantA: this.props.variantA,
-        variantB: this.props.variantB,
-      });
-
       if (!this.isExperimentEnabled()) {
-        log.debug(`Experiment "${props.id}" is not enabled by config.`);
+        log.debug(`Experiment "${defaultId}" is not enabled by config.`);
```
Comment (willdurand) [inline src/core/withExperiment.js]: I believe this should be `props.id`

File: tests/unit/amo/components/TestHomeHeroBanner.js [hunk 3]
```diff
@@ -2,35 +2,18 @@ import * as React from 'react';
...
   fakeI18n,
-  getFakeConfig,
   shallowUntilTarget,
 } from 'tests/unit/helpers';
 
-// Skip `withCookies` HOC since Enzyme does not support the React Context API.
-// See: https://github.com/mozilla/addons-frontend/issues/6839
-jest.mock('react-cookie', () => ({
-  withCookies: (component) => component,
-}));
```
Comment (willdurand) [inline tests/unit/amo/components/TestHomeHeroBanner.js]: 🏆

File: src/amo/components/HomeHeroBanner/index.js (hunk 1 above)
Comment (rebmullin) [reply src/amo/components/HomeHeroBanner/index.js]: oh? So this is only used when using connect/redux?

File: src/core/withExperiment.js (hunk 2 above)
Comment (rebmullin) [reply src/core/withExperiment.js]: oops! I didn't mean to change this.. thanks!

File: src/amo/components/HomeHeroBanner/index.js (hunk 1 above)
Comment (willdurand) [reply src/amo/components/HomeHeroBanner/index.js]: this is only useful when you have more than one HOC

**Pull request merged**